fix: update default node in devstack MFE's to v 18.x

justinhynes opened the pull request:
We are currently installing 16 by default, which is EOL and causing issues with some MFE services in devstack.

davec-edx commented on docker-compose.yml:
```diff
@@ -0,0 +1,96 @@
+      ENABLE_DJANGO_TOOLBAR: 1
+    image: edxops/credentials:latest
+    ports:
+      - "18150:18150"
```
Credentials IDA currently has 18150 and 8150 exposed, is the current one incorrect or is 8150 missing from here?

clintonb replied:
8150 is the `gunicorn` port. 18150 is the `nginx` port. I can't think of any good reason to prefer gunicorn over nginx given that nginx also gives you static assets.

jmbowman commented on docker-compose.yml:
```diff
@@ -0,0 +1,96 @@
+##   that might be running for the same service.
+##
+
+version: "2"
```
Would it be worth using [2.1](https://docs.docker.com/compose/compose-file/#version-21) instead?

clintonb replied:
We aren't using any of those features; but, I see no harm in staying current.

jmbowman commented on Makefile:
```diff
@@ -0,0 +1,13 @@
+.PHONY: devstack.reset devstack.start devstack.stop
+
+devstack.open.%:
```
A help target describing each option would be nice.  I like the way that was done [here](https://github.com/edx/cookiecutter-django-app/blob/master/Makefile#L5), doesn't require any changes as new targets are added

rlucioni replied:
What's up with the percentage symbol here?

Thoughts on using [invoke](http://www.pyinvoke.org/) instead of Make to avoid these questions altogether?

clintonb replied:
> While fully usable, Invoke is still pre-1.0 software and has no backwards compatibility guarantees until the 1.0 release occurs! Please make sure to read the changelog carefully anytime you upgrade!

Besides this, I'm not a big fan of abstracting things too much when it comes to running commands. The farther you get from bash, the more difficult it becomes to debug when things go wrong. The "magic" of edx-platform's Paver setup comes to mind.

jmbowman commented on docker-compose.yml:
```diff
@@ -0,0 +1,96 @@
+    volumes:
+      - ./.dev/volumes/es/data:/usr/share/elasticsearch/data
+      - ./.dev/volumes/es/logs:/usr/share/elasticsearch/logs
+      - ./.dev/volumes/es/plugins:/usr/share/elasticsearch/plugins
```
Would it be worth breaking out the volume definitions for clarity?  May not help that much unless we go with the Compose format 2.1 to get [volume labels](https://docs.docker.com/compose/compose-file/#volume-configuration-reference#labels-1)

clintonb replied:
The volume labels do not appear to correspond to directories on the host. I'll await Renzo's findings on data volumes before making this change.

rlucioni commented on .dev/volumes/.keep:
```diff
@@ -0,0 +1 @@
+This directory contains the data for devstack. The contents of this directory are ignored by Git.
```
Is this actually necessary? I read [Docker's reference](https://docs.docker.com/engine/tutorials/dockervolumes/#data-volumes) on data volumes, and it sounds like mounting host directories isn't necessary to persist volumes even if the container is deleted. For our purposes, mounting of host directories only seems to be necessary for mounting source code inside containers.

clintonb replied:
I have yet to get data volumes to work correctly. For now I'd like to avoid that bit of complexity (given that I can't even get time to make one image!). Once we have a working implementation we can explore areas of improvement.

rlucioni replied:
Okay. I'm still researching this myself, but from the looks of it using the [`VOLUME`](https://docs.docker.com/engine/reference/builder/#/volume) instruction in the Dockerfiles used to build these images may actually simplify things on this end.

clintonb replied:
This has been added to my list of things to explore after we get something up and running.

rlucioni commented on docker-compose.yml:
```diff
@@ -0,0 +1,96 @@
+## This file contains all of the services for an edX installation. See https://docs.docker.com/compose/compose-file/
```
Why the double hash marks? Single should serve the same purpose.

clintonb replied:
Updated.

rlucioni commented on docker-compose.yml:
```diff
@@ -0,0 +1,96 @@
+    ports:
+      - "18150:18150"
+
+  discovery:
```
Don't we need a host volume mounted for our services' source code? You should be able to mount
```
../<repo name>:/path/to/container/source
```

clintonb replied:
We can add that later, once we have things working. I want to avoid introducing sources of complexity too early.

justinhynes committed:
fix: update default node in devstack MFE's to v 18.x

We are currently installing 16 by default, which is EOL and causing issues with some MFE services in devstack.

leangseu-edx reviewed (approved):
LGMT 👍
